src/components/Navigation/AuthenticatedNavigationView.js

```diff
@@ -43,6 +58,63 @@ export default function AuthenticatedNavigationView({ UserNavigationOptions }) {
         setMobileOpen((open) => !open);
     }, []);
 
+    useEffect(() => {
+        if (
+            email !== null &&
+            (google_access_token === null || google_access_token === undefined)
+        ) {
+            setGoogleLoginPromptOpen(true);
+        }
+    }, [email]);
+
+    useEffect(() => {
+        // if(sessionStorage.getItem('google_access_token') == null){
+        //     sessionStorage.setItem(
+        //         'google_access_token',
+        //         google_access_token
+        //     )
+        // }
+        setExitButtonMessage('Exit');
+        if (
+            (gClassResp === null || gClassResp === undefined) &&
+            google_access_token
+        ) {
+            (async () => {
+                try {
+                    setGClassResp(
+                        await axios.get(
+                            'https://classroom.googleapis.com/v1/courses',
+                            {
+                                headers: {
+                                    Authorization: `Bearer ${google_access_token}`,
+                                },
+                            }
+                        )
+                    );
+                } catch (error) {
+                    console.log(error);
+                }
+            })();
+        }
+        if (google_courses === undefined || google_courses == null) {
+            dispatch({
+                type: actions.SET_GOOGLE_COURSES,
+                payload: { google_courses: gClassResp?.data.courses },
+            });
+        }
+    }, [google_access_token]);
+
+    const handleClose = () => {
+        setGoogleLoginPromptOpen(false);
+    };
+
+    const onSuccess = (response) => {
+        // setGoogle_access_token(response.tokenObj.access_token);
```

Comment:
Look into 'response.reloadAuthResponse()'. You can find one implementation here about halfway down the page https://dev.to/sivaneshs/add-google-login-to-your-react-apps-in-10-mins-4del